src/R/Editor/Impl/Validation/Lint/LintOptions.cs

```diff
@@ -1,60 +1,109 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using System;
+using Microsoft.Common.Core.Diagnostics;
+using Microsoft.Languages.Editor.Settings;
+using Microsoft.R.Common.Core.UI;
+using static System.FormattableString;
+
 namespace Microsoft.R.Editor.Validation.Lint {
-    public sealed class LintOptions {
+    public sealed class LintOptions: BindableBase {
+        private readonly Func<IEditorSettingsStorage> _storageAccess;
+        private IEditorSettingsStorage _storage;
+
+        private IEditorSettingsStorage Storage => _storage ?? (_storage = _storageAccess());
+        private IWritableEditorSettingsStorage WritableStorage => Storage as IWritableEditorSettingsStorage;
+
+        public LintOptions(Func<IEditorSettingsStorage> storageAccess) {
+            Check.ArgumentNull(nameof(storageAccess), storageAccess);
+            _storageAccess = storageAccess; // Storage is delay-created after package loads
+        }
+
+        private string GetKey(string optionName) => Invariant($"LintR_{optionName}");
```

Comment:
If you want you can define two methods (`Get` and `Set`) and in this case you don't have to use `nameof`:

```cs
        private T Get<T>(T defaultValue, [CallerMemberName] string name = null) => Storage.Get(GetKey(name), defaultValue);
        private void Set<T>(T value, [CallerMemberName] string name = null) => WritableStorage?.Set(GetKey(name), value);

        public bool Enabled {
            get => Get(false);
            set => Set(value);
        }
```

Up to you though